portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -24,9 +34,79 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-  @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("Hello Laura!");
+  private List<String> messages;
+
+    @Override
+    public void init() {
+      messages = new ArrayList<String>();
+    }
+
+    @Override 
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+      // Converting Array List to JSON.
+      String jsonVersion = convertToJson(messages);
+
+      Query query = new Query("Comments").addSort("comment", SortDirection.ASCENDING);
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+      PreparedQuery results = datastore.prepare(query);
+
+      for (Entity entity : results.asIterable()) {
+        String comment = (String) entity.getProperty("comment");
+
+        Task task = new Task(comment);
+        jsonVersion.add(task);
+    }
+
+      // Send JSON string.
+      response.setContentType("application/json;");
+      response.getWriter().println(jsonVersion);
+    }
+
+    private String convertToJson(List messages) {
+      Gson gson = new Gson();
+      return gson.toJson(messages);
+    }
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+      // Get the input from the form.
+      String text = getParameter(request, "word-input", "");
+      boolean positive = Boolean.parseBoolean(getParameter(request, "positive", "false"));
+      boolean negative = Boolean.parseBoolean(getParameter(request, "negative", "false"));
+      boolean mixed = Boolean.parseBoolean(getParameter(request, "mixed", "false"));
+      
+      Entity taskEntity = new Entity("Comments");
+      taskEntity.setProperty("comment", text);
+
+      // Storing comments in their respective bins.
+      if (positive) {
+        taskEntity.setProperty("status", "positive");
+      } 
+      
+      if (negative) {
+        taskEntity.setProperty("status", "negative");
+      }
+
+      if (mixed) {
+        taskEntity.setProperty("status", "mixed");
+      }
+
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+      datastore.put(taskEntity);
+
+      // Respond with the result.
+      response.setContentType("text/html;");
+      response.getWriter().println(text);
+}
+  /**
+   * @return the request parameter, or the default value if the parameter
+   *         was not specified by the client
+   */
+  private String getParameter(HttpServletRequest request, String name, String defaultValue) {
+    String value = request.getParameter(name);
+    if (value == null) {
```

Comment:
Suggestion: You could use a ternary here for better readability if you want. `return value != null ? value : defaultValue`